crates/polars-expr/src/hot_groups/fixed_index_table.rs

```diff
@@ -73,8 +73,16 @@ impl<K> FixedIndexTable<K> {
         V: FnMut(Q, &mut K),
     {
         let tag = hash as u32;
-        let h1 = (hash >> self.shift) as usize;
-        let h2 = (hash.wrapping_mul(H2_MULT) >> self.shift) as usize;
+        let h1 = if (self.shift as u32) < u64::BITS {
```

Comment:
Please revert this, there should not be branches here. Just assert the shift fits during construction of the table (and also an assert that the table size must be a power of two).